pkg/types/baremetal/validation/platform.go

```diff
@@ -61,6 +61,66 @@ func validateOSImageURI(uri string) error {
 	return nil
 }
 
+// validateHosts checks that hosts have all required fields set with appropriate values
+func validateHosts(hosts []*baremetal.Host, fldPath *field.Path) field.ErrorList {
+	hostErrs := field.ErrorList{}
+
+	uniqueBMC := make(map[string]*baremetal.Host)
+	uniqueMAC := make(map[string]*baremetal.Host)
+	uniqueName := make(map[string]*baremetal.Host)
+
+	for _, host := range hosts {
+		// Ensure name is set
+		if host.Name == "" {
+			hostErrs = append(hostErrs, field.Invalid(fldPath.Child("hosts"), host.Name, "name is required"))
+		}
+
+		// Ensure name is unique
+		if other, ok := uniqueName[host.Name]; ok {
+			hostErrs = append(hostErrs, field.Invalid(fldPath.Child("hosts"), host.Name, fmt.Sprintf("duplicate name %s", other.Name)))
+		} else {
+			uniqueName[host.Name] = host
+		}
+
+		// Ensure MAC is required
+		if host.BootMACAddress == "" {
+			hostErrs = append(hostErrs, field.Invalid(fldPath.Child("hosts"), host.BootMACAddress, "MAC address is required"))
```

Comment:
i think the path here is wrong... similar to https://github.com/openshift/installer/pull/3208/files#r385919052